File: twoSum.go
func twoSum(nums []int, target int) []int {
    indexLocations := make(map[int]int)
    for i := range nums {
        current := nums[i]
        diff := target - current
        elem, ok := indexLocations[diff]
        if ok {
            return []int{elem, i}
        } else {
            indexLocations[current] = i
        }
    }
    return []int{-1, -1}
}

File: two_sum.py
def two_sum(nums, target):
    results = {}
    for i in range(0, len(nums)):
        current = nums[i]
        diff = target - current
        if diff in results:
            return [results[diff], i]
        results[current] = i
    return [-1,-1]

nums = [2, 7, 11, 15]
target = 9
print(two_sum(nums, target))
